### PR TITLE
avoid integer wraparound

### LINKED ISSUE #8056: free(): invalid pointer.  Aborted (core dumped)
Current master  darktable 3.5.0+891~gc672080f5
WSL Ubuntu 20.04

In culling view with focus peaking on.
Was doing a ctrl+w on images in this view and after some time DT crashed.

The terminal had this info

```
free(): invalid pointer
Aborted (core dumped)
```

### LINKED ISSUE #8066: double free or corruption (!pre) - tone equalizer with guided filter
**Describe the bug**
DT crashes with "double free or corruption (!prev)" when loading a file and zooming in and out randomly while having the tone equalizer turned on and using the guided filter.
Sometimes it already crashes when loading a specific images.
Here is a backtrace: [backtrace.txt](https://github.com/darktable-org/darktable/files/5907575/backtrace.txt)

**Platform (please complete the following information):**
 - Darktable Version: lastest master (3.5.0+903~g641608706)
 - OS: Ubuntu 20.04
 - OpenCL activated: yes

I didn't have this crash with the last version I compiled some days ago, I need to check which version I used last time, where I didn't run into this crash.

### PR DESCRIPTION
**Pull request by ralfbrown**
use x+y<z instead of x<z-y to avoid a wrap to very large values when y>z and the variables are unsigned.

Should fix #8056, though I wasn't able to reproduce the crash (couldn't cause a request to blur an image with dimensions smaller than the radius).  Likely to fix #8066 as well, as that seems to be closely related.

### CODE REVIEW

File: src/common/box_filters.c [hunk 1]
```diff
@@ -385,7 +385,7 @@ static void blur_vertical_4ch_sse(float *const restrict buf, const size_t height
   }
   // process the blur up to the point where we start removing values
   size_t y;
-  for (y = 0; y <= MIN(radius, height-1); y++)
+  for (y = 0; y <= MIN(radius, height-radius-1); y++)
```
Comment (TurboGit) [inline src/common/box_filters.c:388]: Can you explain and double check this one ? This is quite a big difference and does not correspond really to the description of the commit. Is that intended ? A typo while editing ? TIA.

**Review by TurboGit — COMMENTED** [review]
One comment, not clear to me...

**Comment by da-phil** [comment]
Fixes https://github.com/darktable-org/darktable/issues/8066

File: src/common/box_filters.c (hunk 1 above)
Comment (ralfbrown) [reply src/common/box_filters.c:388]: This prevents a buffer overrun when 2*radius > height.  There are some other occurrences in the code, but this one was missed previously.  I believe that this change is what fixes #8066.

Comment (TurboGit) [reply src/common/box_filters.c:388]: Understood, thanks.

**Comment by cryptomilk** [comment]
Hey @ralfbrown 

one of your last commits breaks compilation with gcc7.

https://build.opensuse.org/package/live_build_log/graphics:darktable:master/darktable/openSUSE_Leap_15.1/x86_64
https://build.opensuse.org/package/live_build_log/graphics:darktable:master/darktable/xUbuntu_18.04/x86_64

**Comment by cryptomilk** [comment]
To be precise it is a compiler bug in gcc <= 7.4 we are triggering.

**Comment by ralfbrown** [comment]
Unfortunately the log gives me absolutely no information about what might be triggering the compiler crash....

**Comment by cryptomilk** [comment]
I could drop Ubuntu 18.04. For openSUSE 15.1 the update repository was not enabled, after changing to the update repo it got gcc 7.5 which has a fix for the issue.

However we might want to document that `gcc < 7.5` has a compiler bug leading to a compiler segfault.

**Comment by cryptomilk** [comment]
Ok, there is an gcc 7.5 update for 18.04. I've added it we are good again :-)

**Pull request merged**